ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/BatchingQueue.scala

```diff
@@ -69,13 +92,15 @@ case class DefaultBatchingQueue(
       .to(Sink.ignore)
       .run()
 
-    val queueAlive = new AtomicBoolean(true)
-    materializedQueue.watchCompletion.foreach { _ =>
-      queueAlive.set(false)
-    }(materializer.executionContext)
+    val queueState = new AtomicReference[RunningBatchingQueueState](RunningBatchingQueueState.Alive)
+    materializedQueue
+      .watchCompletion()
+      .onComplete { _ =>
+        queueState.compareAndSet(RunningBatchingQueueState.Alive, RunningBatchingQueueState.Failed)
```

Comment:
Shouldn't the state be set to `Complete` instead once the queue is completed?

Reply:
This only happens if it's caused by an exception, not through `stop`. Otherwise the compare and set fails, but the one in `stop` succeeds.